spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestSetWriteDistributionAndOrdering.java

```diff
@@ -279,4 +281,40 @@ public void testSetWriteDistributedAndLocallyOrderedInverted() {
         .build();
     Assert.assertEquals("Sort order must match", expected, table.sortOrder());
   }
+
+  @Test
+  public void testSetWriteOrderByColumnWithCaseInsensitive() {
+    sql("CREATE TABLE %s (id bigint NOT NULL, category string, ts timestamp, data string) USING iceberg", tableName);
+    Table table = validationCatalog.loadTable(tableIdent);
+    Assert.assertTrue("Table should start unsorted", table.sortOrder().isUnsorted());
+
+    spark.conf().set("spark.sql.caseSensitive", "false");
+    sql("ALTER TABLE %s WRITE ORDERED BY CATEGORY, Id", tableName);
+
+    table.refresh();
+
+    String distributionMode = table.properties().get(TableProperties.WRITE_DISTRIBUTION_MODE);
+    Assert.assertEquals("Distribution mode must match", "range", distributionMode);
+
+    SortOrder expected = SortOrder.builderFor(table.schema())
+            .withOrderId(1)
+            .asc("category", NullOrder.NULLS_FIRST)
+            .asc("id", NullOrder.NULLS_FIRST)
+            .build();
+    Assert.assertEquals("Should have expected order", expected, table.sortOrder());
+  }
+
+  @Test
+  public void testSetWriteOrderByColumnWithCaseSensitive() {
+    sql("CREATE TABLE %s (id bigint NOT NULL, category string, ts timestamp, data string) USING iceberg", tableName);
+    Table table = validationCatalog.loadTable(tableIdent);
+    Assert.assertTrue("Table should start unsorted", table.sortOrder().isUnsorted());
+
+    spark.conf().set("spark.sql.caseSensitive", "true");
+    AssertHelpers.assertThrows("Should reject invalid `write ordered` columns",
+            ValidationException.class, "Cannot find field 'CATEGORY' in struct",
+        () -> {
```

Comment:
Nit: believe this lambda could fit on one line and there’s no need for the brackets.